cmd/api/src/api/v2/savedqueriespermissions.go

```diff
@@ -0,0 +1,198 @@
+				}
+			}
+		}
+	}
+}
+
+// DeleteSavedQueryPermissionsRequest represents the payload sent to the unshare endpoint
+type DeleteSavedQueryPermissionsRequest struct {
+	UserIds []uuid.UUID `json:"user_ids"`
+}
+
+// DeleteSavedQueryPermissions allows an owner of a shared query, a user that has a saved query shared to them, or an admin, to remove sharing privileges.
+// A user who owns a query may unshare a query from anyone they have shared to
+// A user who had a query shared to them may unshare that query from themselves
+// And admins may unshare queries that have been shared to other users
+func (s Resources) DeleteSavedQueryPermissions(response http.ResponseWriter, request *http.Request) {
+	var (
+		rawSavedQueryID = mux.Vars(request)[api.URIPathVariableSavedQueryID]
+		deleteRequest   DeleteSavedQueryPermissionsRequest
+	)
+
+	if user, isUser := auth.GetUserFromAuthCtx(ctx2.FromRequest(request).AuthCtx); !isUser {
+		api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusBadRequest, "No associated user found", request), response)
+	} else if savedQueryID, err := strconv.ParseInt(rawSavedQueryID, 10, 64); err != nil {
+		api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusBadRequest, api.ErrorResponseDetailsFromMalformed, request), response)
+	} else if err = json.NewDecoder(request.Body).Decode(&deleteRequest); err != nil {
+		api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusBadRequest, err.Error(), request), response)
+	} else {
+		// Check if the user is attempting to unshare a query from themselves
+		if slices.Contains(deleteRequest.UserIds, user.ID) {
+			if isShared, err := s.DB.IsSavedQuerySharedToUser(request.Context(), savedQueryID, user.ID); err != nil {
+				api.HandleDatabaseError(request, response, err)
+				return
+			} else if !isShared {
+				// The user cannot unshare a saved query if a saved query permission does not exist for them. This means a user cannot unshare a query that they don't own, or hasn't been shared with them
+				api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusBadRequest, "User cannot unshare a query from themselves that is not shared to them", request), response)
+				return
+			}
+		} else {
+			// User is attempting to unshare a query from another user
+			isAdmin := user.Roles.Has(model.Role{Name: auth.RoleAdministrator})
+			if !isAdmin {
+				// If a user is not admin, then they need to own the query in order to unshare it
+				if savedQueryBelongsToUser, err := s.DB.SavedQueryBelongsToUser(request.Context(), user.ID, savedQueryID); err != nil {
+					api.HandleDatabaseError(request, response, err)
+					return
+				} else if !savedQueryBelongsToUser {
+					api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusForbidden, "Query does not belong to the user", request), response)
+					return
+				}
```

Comment:
This was the only thing that was changed besides moving your function @mvlipka! 

It got switched from 401 to 403 since according to @mistahj67, there are some handlers on the frontend that will automatically logout a user whenever they receive a 401. And we don't want them to get logged out in this scenario